AppCenter/AppCenter/Internals/Model/Util/MSUserDefaults.m

```diff
@@ -1,42 +1,84 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#import "MSUserDefaults.h"
 #import "MSAppCenterInternal.h"
 #import "MSLogger.h"
-#import "MSUserDefaults.h"
 
 static NSString *const kMSUserDefaultsTs = @"_ts";
+static NSString *const kMSAppCenterUserDefaultsMigratedKeyFormat = @"%@UserDefaultsMigratedKey";
+static NSString *const kMSUserDefaultsPrefix = @"MSAC";
+
+static MSUserDefaults *sharedInstance = nil;
+static dispatch_once_t onceToken;
 
 @implementation MSUserDefaults
 
 + (instancetype)shared {
-  static MSUserDefaults *sharedInstance = nil;
-  static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
     sharedInstance = [[MSUserDefaults alloc] init];
+    NSDictionary *migratedKeys = @{
+      @"MSChannelStartTimer" : @"MSACChannelStartTimer",
+      @"pastDevicesKey" : @"MSACPastDevicesKey",
+      @"MSInstallId" : @"MSACInstallId",
+      @"MSAppCenterIsEnabled" : @"MSACAppCenterIsEnabled",
+      @"MSEncryptionKeyMetadata" : @"MSACEncryptionKeyMetadata",
+      @"SessionIdHistory" : @"MSACSessionIdHistory",
+      @"UserIdHistory" : @"MSACUserIdHistory"
+    };
+    [sharedInstance migrateKeys:migratedKeys forService:@"Core"];
   });
   return sharedInstance;
 }
 
++ (void)resetSharedInstance {
+  onceToken = 0; // resets the once_token so dispatch_once will run again
+  sharedInstance = nil;
+}
+
+- (void)migrateKeys:(NSDictionary *)migratedKeys forService:(NSString *)serviceName {
+  NSString *serviceHasMigratedKey = [NSString stringWithFormat:kMSAppCenterUserDefaultsMigratedKeyFormat, serviceName];
+  NSNumber *serviceHasMigrated = [self objectForKey:serviceHasMigratedKey];
+  if (serviceHasMigrated) {
```

Comment:
Let's add a verbose log here that says it has already been migrated.

Reply:
Are you sure about it? It means user will get this log 99% of the app starts (1 time it has migrated and all the subsequent starts will contain this log)

Reply:
Oops. I missed L45. It should be good then.